docker/base/build.sh

```diff
@@ -176,12 +176,16 @@ if [ "$TARGETS" == "" ]; then
   TARGETS="./."
 fi
 
+if [ -f "$PACK_RELPATH/.xgo/setup.sh" ]; then echo "source setup.sh"; source "$PACK_RELPATH/.xgo/setup.sh"; fi
+
 # Build for each requested platform individually
 for TARGET in $TARGETS; do
   # Split the target into platform and architecture
   XGOOS=`echo $TARGET | cut -d '/' -f 1`
   XGOARCH=`echo $TARGET | cut -d '/' -f 2`
 
+  if [ -f "$PACK_RELPATH/.xgo/build.sh" ]; then echo "source build.sh"; source "$PACK_RELPATH/.xgo/build.sh"; fi
```

Comment:
`build_deps.sh` may be a better location for this, as it is where `./configure` is also called.

Reply:
it may be but AFAIK in `build_deps.sh` the `XGOOS` and `XGOARCH` variables (as well as all other variables) are not visible.

Reply:
@techknowlogick how should we proceed here?

Reply:
So I still don't know that this will work, as say if I want to build only for windows I would set my target via CLI to be `windows/*` which will translate into the variable `TARGET` in the loop to be `windows/.` and so `.xgo/build.sh` won't run for each platform.

Reply:
I think what @techknowlogick is trying to say is that if TARGETS="./." - this build.sh will only be run once for all archs. I don't think that can be the intention.

Reply:
Indeed, and I'll post your suggestion from chat: I think it might be better to have two loops run on  TARGETS a first that rewrites it  expands out wildcards - then you can simplify the second loop and this PR is right

Reply:
I see. instead or having two loops (in my opinion more error prone), the script is now called in a separate function which also calls build_dep.sh. XGOxx vars are expanded for the actual platform